physics/kepler_orbit_body.hpp

```diff
@@ -157,94 +213,478 @@ KeplerOrbit<Frame>::KeplerOrbit(
+  } else if (longitude_of_periapsis) {
+    auto const& ϖ = *longitude_of_periapsis;
+    argument_of_periapsis = ϖ - Ω;
+  }
+}
+
+template<typename Frame>
+void KeplerOrbit<Frame>::CompleteElements(KeplerianElements<Frame>& elements,
+                                          GravitationalParameter const& μ) {
+  CompleteConicParameters(elements, μ);
+  CompleteOrientationParameters(elements);
+  CompleteAnomalies(elements);
+}
+
+template<typename Frame>
+void KeplerOrbit<Frame>::CompleteAnomalies(KeplerianElements<Frame>& elements) {
+  auto const& e = *elements.eccentricity;
+  auto& true_anomaly = elements.true_anomaly;
+  auto& mean_anomaly = elements.mean_anomaly;
+  auto& hyperbolic_mean_anomaly = elements.hyperbolic_mean_anomaly;
+  CHECK_EQ(true_anomaly.has_value() + mean_anomaly.has_value() +
+               hyperbolic_mean_anomaly.has_value(),
+           1);
+  if (true_anomaly) {
+    auto const& ν = *true_anomaly;
+    auto const positive_angle = [](Angle const& α) -> Angle {
+      return α > 0 * Radian ? α : α + 2 * π * Radian;
+    };
+    Angle const eccentric_anomaly =
+        ArcTan(Sqrt(1 - Pow<2>(e)) * Sin(ν), e + Cos(ν));
+    mean_anomaly =
+        positive_angle(eccentric_anomaly - e * Sin(eccentric_anomaly) * Radian);
+    Angle const hyperbolic_eccentric_anomaly =
+        ArcCosh((e + Cos(ν)) / (1 + e * Cos(ν)));
+    hyperbolic_mean_anomaly = e * Sinh(hyperbolic_eccentric_anomaly) * Radian -
+                              hyperbolic_eccentric_anomaly;
+  } else if (mean_anomaly) {
+    auto const kepler_equation =
+        [e, mean_anomaly](Angle const& eccentric_anomaly) -> Angle {
+      return *mean_anomaly -
+             (eccentric_anomaly - e * Sin(eccentric_anomaly) * Radian);
+    };
+    Angle const eccentric_anomaly = e == 0 ? *mean_anomaly
+                                           : Bisect(kepler_equation,
+                                                    *mean_anomaly - e * Radian,
+                                                    *mean_anomaly + e * Radian);
+    true_anomaly = 2 * ArcTan(Sqrt(1 + e) * Sin(eccentric_anomaly / 2),
+                              Sqrt(1 - e) * Cos(eccentric_anomaly / 2));
+    hyperbolic_mean_anomaly = NaN<Angle>();
+  } else if (hyperbolic_mean_anomaly) {
+    auto const hyperbolic_kepler_equation =
+        [e, hyperbolic_mean_anomaly](
+            Angle const& hyperbolic_eccentric_anomaly) -> Angle {
+      return *hyperbolic_mean_anomaly -
+             (e * Sinh(hyperbolic_eccentric_anomaly) * Radian -
+              hyperbolic_eccentric_anomaly);
+    };
+    Angle const hyperbolic_eccentric_anomaly =
+        Bisect(hyperbolic_kepler_equation,
+               0 * Radian,
+               *hyperbolic_mean_anomaly / (e - 1));
+    true_anomaly =
+        2 * ArcTan(Sqrt(e + 1) * Sinh(hyperbolic_eccentric_anomaly / 2),
+                   Sqrt(e - 1) * Cosh(hyperbolic_eccentric_anomaly / 2));
+    mean_anomaly = -NaN<Angle>();
```

Comment:
It is there for consistency with the result of the computations from the true anomaly. This will become saner once we have Radical, for now it makes it much easier to test.